kubernetes/cmsweb/crons/cron-cric.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: cric-pv
+spec:
+  capacity:
+    storage: 1Gi
+  accessModes:
+    - ReadWriteMany
+  hostPath:
+    path: "/etc/cric"  
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: cric-pvc
+  namespace: auth
+spec:
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 1Gi
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: cron-cric
+  namespace: auth
+spec:
+  schedule: "0 * * * *"  # Runs every hour
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          securityContext:
+            runAsUser: 0
+            runAsGroup: 0
+            fsGroup: 0
+          containers:
+          - name: cric-fetch
+            image: registry.cern.ch/cmsweb/curl:latest-stable
+            args:
+            - /bin/sh
+            - -c
+            - >
+              curl --cert /etc/robots/robotcert.pem --key /etc/robots/robotkey.pem -k -L -s "https://cms-cric.cern.ch/api/accounts/user/query/?json&preset=roles" -o /etc/cric/cric.json
```

Comment:
Aroosha, this is not correct as CRIC service may go down and curl command may timeout or fail. We must account for errors. Therefore I suggest that you expand this command to a script where you'll check for status of last command and only then overwrite /etc/cric/cric.json

For example, something like this:
```
#!/bin/bash
odir=/etc/cric/cric.json
tstamp=`date "+%Y%M%d_%T"`
curl --cert /etc/robots/robotcert.pem --key /etc/robots/robotkey.pem \
    -k -L -s -w "%{response_code}" \
    "https://cms-cric.cern.ch/api/accounts/user/query/?json&preset=roles" -o ${odir}.${tstamp}
if [ "$?" == "0" ] && [ "$response_code" == 200 ]; then
    mv ${odir}.${tstamp} ${odir}
fi
```
Please check this script thoroughly and make it fail, e.g. change `roles` to `roles2` (or any other value which will cause curl call to fail). Here I'm asking curl to return HTTP response code and verify that it must be 200 along with correct unix code.

Reply:
Ah, thanks for the explanation! Now I understand the usecase.

Do you mean something like this:

```
[apervaiz@lxplus823 cmsweb]$ k logs cron-cric-28746214-h669f -n auth
Starting CRIC fetch script...
Curl command executed with response code: 200
CRIC data fetched successfully and moved to /etc/cric/cric.json.
=== Head of /etc/cric/cric.json ===
[
  {
    "DN": "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=xge/CN=874768/CN=Xian Ge",
    "EMAIL": "07220231@njnu.edu.cn",
    "ID": 874768,
    "LOGIN": "xge",
    "NAME": "Xian Ge",
    "ROLES": {
      "user": [
        "group:users"
=== Tail of /etc/cric/cric.json ===
    "ID": 865564,
    "LOGIN": "zixiong",
    "NAME": "Ziv Zhou",
    "ROLES": {
      "user": [
        "group:users"
      ]
    }
  }
]Script completed.
[apervaiz@lxplus823 cmsweb]$
```

Reply:
After testing the script.. If I make changes that are wrong:

```
[apervaiz@lxplus823 cmsweb]$ k logs cron-cric-28746230-5sh5g -n auth
Starting CRIC fetch script...
Curl command executed with response code: 400
Failed to fetch CRIC data or invalid response code: 400
=== Head of /etc/cric/cric.json ===
[
  {
    "DN": "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=xge/CN=874768/CN=Xian Ge",
    "EMAIL": "07220231@njnu.edu.cn",
    "ID": 874768,
    "LOGIN": "xge",
    "NAME": "Xian Ge",
    "ROLES": {
      "user": [
        "group:users"
=== Tail of /etc/cric/cric.json ===
    "ID": 865564,
    "LOGIN": "zixiong",
    "NAME": "Ziv Zhou",
    "ROLES": {
      "user": [
        "group:users"
      ]
    }
  }
]Script completed.
[apervaiz@lxplus823 cmsweb]$

```

Reply:
yes, exactly, but to avoid confusion when curl fails you should put message that the content of `/etc/cric/cric.json` is not modified and provide instead content of output from failed curl. Please make further adjustment to show that.